app/code/Magento/Payment/Model/Method/Adapter.php

```diff
@@ -31,7 +32,7 @@
  * @api Use this class as a base for virtual types declaration
  * @since 100.0.2
  */
-class Adapter implements MethodInterface, SaleOperationInterface
+class Adapter implements MethodInterface, SaleOperationInterface, PaymentMethodInterface
```

Comment:
There is already such methods duplication in 

```
abstract class AbstractMethod extends \Magento\Framework\Model\AbstractExtensibleModel implements
    MethodInterface,
    PaymentMethodInterface
```

is there any real issue observable due to
> would technically not be valid due to the missing interface

?

From a quick code review it looks like `PaymentMethodInterface` is what `Quote` needs to know about payment methods in its bounded context, there is even another similar interface in `Payment` module itself `\Magento\Payment\Api\Data\PaymentMethodInterface`.

My understanding is that Quote should be aware of Payment and not vice versa.